scipy/stats/_censored_data.py

```diff
@@ -0,0 +1,367 @@
+                f" intervals={intervals_str})")
+
+    def __str__(self):
+        num_nc = len(self._x)
+        num_lc = len(self._left)
+        num_rc = len(self._right)
+        num_ic = len(self._intervals)
+        n = num_nc + num_lc + num_rc + num_ic
+        parts = [f'{num_nc} not censored']
+        if num_lc > 0:
+            parts.append(f'{num_lc} left-censored')
+        if num_rc > 0:
+            parts.append(f'{num_rc} right-censored')
+        if num_ic > 0:
+            parts.append(f'{num_ic} interval-censored')
+        return f'CensoredData({n} values: ' + ', '.join(parts) + ')'
+
+    # This is not a complete implementation of the arithmetic operators.
+    # All we need is subtracting a scalar and dividing by a scalar.
+
+    def __sub__(self, other):
+        return CensoredData(x=self._x - other,
+                            left=self._left - other,
+                            right=self._right - other,
+                            intervals=self._intervals - other)
+
+    def __truediv__(self, other):
+        return CensoredData(x=self._x / other,
+                            left=self._left / other,
+                            right=self._right / other,
+                            intervals=self._intervals / other)
+
+    def __len__(self):
+        """
+        The number of values (censored and not censored).
+        """
+        return (len(self._x) + len(self._left) + len(self._right)
+                + len(self._intervals))
+
+    def num_censored(self):
+        """
+        Number of censored values.
+        """
+        return len(self._left) + len(self._right) + len(self._intervals)
+
+    @classmethod
+    def right_censored(cls, x, censored):
+        """
+        Create a `CensoredData` instance of right-censored data.
+
+        Parameters
+        ----------
+        x : array_like
+            `x` is the array of observed data or measurements.
+            `x` must be a one-dimensional sequence of finite numbers.
+        censored : array_like of bool
+            `censored` must be a one-dimensional sequence of boolean
+            values.  If ``censored[k]`` is True, the corresponding value
+            in `x` is right-censored.  That is, the value ``x[k]``
+            is the lower bound of the true (but unknown) value.
+
+        Returns
+        -------
+        data : `CensoredData`
+            An instance of `CensoredData` that represents the
+            collection of uncensored and right-censored values.
+
+        Examples
+        --------
+        >>> from scipy.stats import CensoredData
+
+        Two uncensored values (4 and 10) and two right-censored values
+        (24 and 25).
+
+        >>> data = CensoredData.right_censored([4, 10, 24, 25],
+        ...                                    [False, False, True, True])
+        >>> data
+        CensoredData(x=array([ 4., 10.]), left=array([], dtype=float64),
+        right=array([24., 25.]), intervals=array([], shape=(0, 2),
```

Comment:
We can consider not printing info about `left, right, ...` if the arrays are empty.

Reply:
Presumably this goes back to  `__repr__` including all the detail needed to re-create the object.